MirrorProvider/MirrorProvider.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -316,6 +320,22 @@ private void OnFileModifiedOrDeleted(string relativePath, bool isDirectory, bool
             Console.WriteLine($"OnFileModifiedOrDeleted: `{relativePath}`, isDirectory: {isDirectory}, isModfied: {isFileDeleted}, isDeleted: {isFileDeleted}");
         }
 
+        private void OnFileRenamed(
+            string relativePath,
+            string relativeDestinationPath,
+            bool isDirectory,
+            ref NotificationType notificationMask)
+        {
+            Console.WriteLine($"OnFileRenamed (isDirectory: {isDirectory}), relativePath: {relativePath}, relativeDestinationPath: {relativeDestinationPath}");
+        }
+
+        private void OnHardlinkCreated(
+            string relativePath,

```

Comment:
Same question as elsewhere - there's no guarantee this is relative, right?

Reply:
Er, for one of these two parameters. I actually can't tell which one is the path of the link and which one is the path of the target :-)

Reply:
> Same question as elsewhere - there's no guarantee this is relative, right?

As discussed elsewhere, any paths outside of the root will be an empty string (by design)

> I actually can't tell which one is the path of the link and which one is the path of the target :-)

Same comment here too, do you think we should use our own names or stay consistent with ProjFS?